### PR TITLE
16 reset password

### LINKED ISSUE #16: reset password

### PR DESCRIPTION
**Pull request by danielbubanec**

### CODE REVIEW

File: src/components/ResetPassword/ResetPassword.jsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  const initialResponse = { message: '', alertVariant: '' };
+  const [response, setResponse] = useState(initialResponse);
+  const [email, setEmail] = useState('')
+
+  const handleSubmit = async e => {
+    e.preventDefault();
+    try {
+      const res = await AuthService.initializePasswordReset({email}) 
+      if(res.status === 'Success'){
+        setResponse(response => ({...response, message: res.message, alertVariant: 'success'}));
```
Comment (slavonicsniper) [inline src/components/ResetPassword/ResetPassword.jsx:15]: ...response not needed

File: src/components/ResetPassword/ResetPassword.jsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      } else{
+        setResponse(response => ({...response, message: res.message, alertVariant: 'danger'}));
```
Comment (slavonicsniper) [inline src/components/ResetPassword/ResetPassword.jsx:17]: ...response not needed

File: src/components/ResetPassword/ResetPassword.jsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      }
+    } catch(error) {
+      console.log(error)
+    }
+  }
+
+  return (
+    <div>
+      <br></br>
+        <Container>
```
Comment (slavonicsniper) [inline src/components/ResetPassword/ResetPassword.jsx:27]: should be centered vertically and horizontally

File: src/components/ResetPasswordConfirmation/ResetPasswordConfirmation.jsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
+import React, {useState, useEffect} from 'react'
+import AuthService from '../../services/AuthService';
+import {Alert, Container, Form, Button} from 'react-bootstrap'
+import { useParams } from "react-router-dom";
+
+function ResetPasswordConfirmation() {
+    let params = useParams();
+    const initialResponse = { message: '', alertVariant: '' };
```
Comment (slavonicsniper) [inline src/components/ResetPasswordConfirmation/ResetPasswordConfirmation.jsx:8]: could be included directly on line 10 when defining response state

File: src/components/ResetPasswordConfirmation/ResetPasswordConfirmation.jsx [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    const [password, setPassword] = useState('')
+    const [response, setResponse] = useState(initialResponse);
+
+    const handleSubmit = async e => {
+        e.preventDefault();
+        try {
+            const res = await AuthService.verifyPasswordReset(params.confirmationCode, {password}) 
+            console.log(res)
```
Comment (slavonicsniper) [inline src/components/ResetPasswordConfirmation/ResetPasswordConfirmation.jsx:16]: probably remove?

File: src/components/ResetPasswordConfirmation/ResetPasswordConfirmation.jsx [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+            if(res.status === 'Success'){
+              setResponse(response => ({...response, message: res.message, alertVariant: 'success'}));
```
Comment (slavonicsniper) [inline src/components/ResetPasswordConfirmation/ResetPasswordConfirmation.jsx:18]: we do not need ...response

File: src/components/ResetPasswordConfirmation/ResetPasswordConfirmation.jsx [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+            } else{
+              setResponse(response => ({...response, message: res.message, alertVariant: 'danger'}));
```
Comment (slavonicsniper) [inline src/components/ResetPasswordConfirmation/ResetPasswordConfirmation.jsx:20]: we do not need ...response

File: src/components/ResetPasswordConfirmation/ResetPasswordConfirmation.jsx [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+                {response.message && 
+                <Alert variant={response.alertVariant}>
+                    {response.message}
+                </Alert>
+                }
+                <Form onSubmit={handleSubmit}>
+                    <h1>Enter new password</h1>
+                    <Form.Group className="mb-3" controlId="formProfileEmail">
+                    <Form.Label>Password</Form.Label>
+                    <Form.Control type="password" placeholder="password"/>
```
Comment (slavonicsniper) [inline src/components/ResetPasswordConfirmation/ResetPasswordConfirmation.jsx:40]: missing indentation

File: src/components/ResetPasswordConfirmation/ResetPasswordConfirmation.jsx [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+                    </Form.Group>
+                    <Form.Group className="mb-3" controlId="formProfilePassword">
+                    <Form.Label>Confirm password</Form.Label>
+                    <Form.Control type="password" placeholder="confirm password" onChange={e => setPassword(e.target.value)}/>
```
Comment (slavonicsniper) [inline src/components/ResetPasswordConfirmation/ResetPasswordConfirmation.jsx:44]: missing indentation

File: src/components/ResetPasswordConfirmation/ResetPasswordConfirmation.jsx [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+    return (
+        <div>
+            <br></br>
+            <Container>
+                {response.message && 
+                <Alert variant={response.alertVariant}>
+                    {response.message}
+                </Alert>
+                }
+                <Form onSubmit={handleSubmit}>
```
Comment (slavonicsniper) [inline src/components/ResetPasswordConfirmation/ResetPasswordConfirmation.jsx:36]: we need to use formik for validation

File: src/services/AuthService.jsx [hunk 11]
```diff
@@ -77,10 +77,62 @@ const verifyUser = async (code) => {
...
+            } else {
+                throw Error(response.statusText);
+            }
+        })
+        .then(async (result) => {
+                return result;
+            },
+            (error) => {
+                return error;
+        });
```
Comment (slavonicsniper) [inline src/services/AuthService.jsx:101]: preferably using catch without `then`

File: src/services/AuthService.jsx [hunk 12]
```diff
@@ -77,10 +77,62 @@ const verifyUser = async (code) => {
...
+    const requestOptions = {
+        method: 'PUT',
+        headers: { 
+            'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(body),
+    };
+    let resData = await fetch(process.env.REACT_APP_API_URL + '/users/reset-password/' + code, requestOptions)
+        .then(response => {
+            if (response.status >= 200 && response.status <= 299) {
```
Comment (slavonicsniper) [inline src/services/AuthService.jsx:115]: this is handled with `response.ok`

**Review by erikgalajda — CHANGES_REQUESTED** [review]

**Review by erikgalajda — APPROVED** [review]
@danoblinux well done and thank you!

**Pull request merged**